### PR TITLE
GenomicsDB matches CombineGVCFs with input spanning deletions

### LINKED ISSUE #5045: GenomicsDBImport: A fatal error has been detected by the Java Runtime Environment
Hi,

I am using GATK `version gatk4-4.0.6.0-0` as part of the bcbio-nextgen pipeline for RNA-seq variant calling. There is one step in the pipeline i.e. `gatk GenomicsDBImport` that's been failing consistently no matter how less or many resources in terms of memory and cores I provide. I have tried to run the command as part of the pipeline and in stand-alone mode (like below) and both produce the same error:

```
[rathik@reslnrefo01 log]$ gatk --java-options '-Xms454m -Xmx3181m -XX:+UseSerialGC' GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path CDL-164-04P-1_0_249250621_genomicsdb -L 1:1-249250621 --variant /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/variation/rnaseq/gatk-haplotype/Sample_1__CDL-164-04P-gatk-haplotype-annotated-rnaedit-annotated-gemini.vcf.gz
Using GATK jar /mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar
Running:
    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xms454m -Xmx3181m -XX:+UseSerialGC -jar /mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path CDL-164-04P-1_0_249250621_genomicsdb -L 1:1-249250621 --variant /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/variation/rnaseq/gatk-haplotype/Sample_1__CDL-164-04P-gatk-haplotype-annotated-rnaedit-annotated-gemini.vcf.gz
Picked up _JAVA_OPTIONS: -Djava.io.tmpdir=/mnt/isilon/cbmi/variome/tmp/rathik
11:49:24.784 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar!/com/intel/gkl/native/libgkl_compression.so
11:49:25.130 INFO  GenomicsDBImport - ------------------------------------------------------------
11:49:25.130 INFO  GenomicsDBImport - The Genome Analysis Toolkit (GATK) v4.0.6.0
11:49:25.131 INFO  GenomicsDBImport - For support and documentation go to https://software.broadinstitute.org/gatk/
11:49:25.134 INFO  GenomicsDBImport - Executing as rathik@reslnrefo01.research.chop.edu on Linux v3.10.0-862.el7.x86_64 amd64
11:49:25.134 INFO  GenomicsDBImport - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_171-b10
11:49:25.134 INFO  GenomicsDBImport - Start Date/Time: July 20, 2018 11:49:24 AM EDT
11:49:25.134 INFO  GenomicsDBImport - ------------------------------------------------------------
11:49:25.134 INFO  GenomicsDBImport - ------------------------------------------------------------
11:49:25.135 INFO  GenomicsDBImport - HTSJDK Version: 2.16.0
11:49:25.135 INFO  GenomicsDBImport - Picard Version: 2.18.7
11:49:25.136 INFO  GenomicsDBImport - HTSJDK Defaults.COMPRESSION_LEVEL : 2
11:49:25.136 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false
11:49:25.136 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true
11:49:25.136 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false
11:49:25.136 INFO  GenomicsDBImport - Deflater: IntelDeflater
11:49:25.136 INFO  GenomicsDBImport - Inflater: IntelInflater
11:49:25.136 INFO  GenomicsDBImport - GCS max retries/reopens: 20
11:49:25.136 INFO  GenomicsDBImport - Using google-cloud-java patch 6d11bef1c81f885c26b2b56c8616b7a705171e4f from https://github.com/droazen/google-cloud-java/tree/dr_all_nio_fixes
11:49:25.137 INFO  GenomicsDBImport - Initializing engine
11:49:25.926 INFO  IntervalArgumentCollection - Processing 249250621 bp from intervals
11:49:25.931 INFO  GenomicsDBImport - Done initializing engine
Created workspace /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/log/CDL-164-04P-1_0_249250621_genomicsdb
11:49:26.339 INFO  GenomicsDBImport - Vid Map JSON file will be written to CDL-164-04P-1_0_249250621_genomicsdb/vidmap.json
11:49:26.339 INFO  GenomicsDBImport - Callset Map JSON file will be written to CDL-164-04P-1_0_249250621_genomicsdb/callset.json
11:49:26.339 INFO  GenomicsDBImport - Complete VCF Header will be written to CDL-164-04P-1_0_249250621_genomicsdb/vcfheader.vcf
11:49:26.339 INFO  GenomicsDBImport - Importing to array - CDL-164-04P-1_0_249250621_genomicsdb/genomicsdb_array
11:49:26.339 INFO  ProgressMeter - Starting traversal
11:49:26.340 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Batches Processed   Batches/Minute
11:49:26.697 INFO  GenomicsDBImport - Importing batch 1 with 1 samples
Buffer resized from 22726bytes to 32529
Buffer resized from 32529bytes to 32693
Buffer resized from 32693bytes to 32738
Buffer resized from 32738bytes to 32741
Buffer resized from 32741bytes to 32756
Buffer resized from 32756bytes to 32768
Buffer resized from 32768bytes to 32769
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f9935bac359, pid=8320, tid=0x00007f99881bf700
#
# JRE version: OpenJDK Runtime Environment (8.0_171-b10) (build 1.8.0_171-b10)
# Java VM: OpenJDK 64-Bit Server VM (25.171-b10 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libtiledbgenomicsdb1661720680664773125.so+0x155359]  BufferVariantCell::set_cell(void const*)+0x99
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/log/hs_err_pid8320.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

Then, I tried to enable core dumping:

```
[rathik@reslnrefo01 CDL-164-04P]$ ulimit -c unlimited

[rathik@reslnrefo01 CDL-164-04P]$ gatk --java-options '-Xms454m -Xmx3181m -XX:+UseSerialGC -Djava.io.tmpdir=/mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/joint/gatk-haplotype-joint/CDL-164-04P/1/bcbiotx/tmp1SAfjS' GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path CDL-164-04P-1_0_249250621_genomicsdb -L 1:1-249250621 --variant /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/variation/rnaseq/gatk-haplotype/Sample_1__CDL-164-04P-gatk-haplotype-annotated-rnaedit-annotated-gemini.vcf.gz
Using GATK jar /mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar
Running:
    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xms454m -Xmx3181m -XX:+UseSerialGC -Djava.io.tmpdir=/mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/joint/gatk-haplotype-joint/CDL-164-04P/1/bcbiotx/tmp1SAfjS -jar /mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path CDL-164-04P-1_0_249250621_genomicsdb -L 1:1-249250621 --variant /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/variation/rnaseq/gatk-haplotype/Sample_1__CDL-164-04P-gatk-haplotype-annotated-rnaedit-annotated-gemini.vcf.gz
Picked up _JAVA_OPTIONS: -Djava.io.tmpdir=/mnt/isilon/cbmi/variome/tmp/rathik
10:24:55.378 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar!/com/intel/gkl/native/libgkl_compression.so
10:24:56.133 INFO  GenomicsDBImport - ------------------------------------------------------------
10:24:56.133 INFO  GenomicsDBImport - The Genome Analysis Toolkit (GATK) v4.0.6.0
10:24:56.134 INFO  GenomicsDBImport - For support and documentation go to https://software.broadinstitute.org/gatk/
10:24:56.136 INFO  GenomicsDBImport - Executing as rathik@reslnrefo01.research.chop.edu on Linux v3.10.0-862.9.1.el7.x86_64 amd64
10:24:56.136 INFO  GenomicsDBImport - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_171-b10
10:24:56.136 INFO  GenomicsDBImport - Start Date/Time: July 23, 2018 10:24:55 AM EDT
10:24:56.136 INFO  GenomicsDBImport - ------------------------------------------------------------
10:24:56.136 INFO  GenomicsDBImport - ------------------------------------------------------------
10:24:56.137 INFO  GenomicsDBImport - HTSJDK Version: 2.16.0
10:24:56.138 INFO  GenomicsDBImport - Picard Version: 2.18.7
10:24:56.138 INFO  GenomicsDBImport - HTSJDK Defaults.COMPRESSION_LEVEL : 2
10:24:56.138 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false
10:24:56.138 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true
10:24:56.138 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false
10:24:56.138 INFO  GenomicsDBImport - Deflater: IntelDeflater
10:24:56.138 INFO  GenomicsDBImport - Inflater: IntelInflater
10:24:56.139 INFO  GenomicsDBImport - GCS max retries/reopens: 20
10:24:56.139 INFO  GenomicsDBImport - Using google-cloud-java patch 6d11bef1c81f885c26b2b56c8616b7a705171e4f from https://github.com/droazen/google-cloud-java/tree/dr_all_nio_fixes
10:24:56.139 INFO  GenomicsDBImport - Initializing engine
10:24:57.198 INFO  IntervalArgumentCollection - Processing 249250621 bp from intervals
10:24:57.205 INFO  GenomicsDBImport - Done initializing engine
Created workspace /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/CDL-164-04P-1_0_249250621_genomicsdb
10:24:57.553 INFO  GenomicsDBImport - Vid Map JSON file will be written to CDL-164-04P-1_0_249250621_genomicsdb/vidmap.json
10:24:57.554 INFO  GenomicsDBImport - Callset Map JSON file will be written to CDL-164-04P-1_0_249250621_genomicsdb/callset.json
10:24:57.554 INFO  GenomicsDBImport - Complete VCF Header will be written to CDL-164-04P-1_0_249250621_genomicsdb/vcfheader.vcf
10:24:57.554 INFO  GenomicsDBImport - Importing to array - CDL-164-04P-1_0_249250621_genomicsdb/genomicsdb_array
10:24:57.554 INFO  ProgressMeter - Starting traversal
10:24:57.554 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Batches Processed   Batches/Minute
10:24:57.971 INFO  GenomicsDBImport - Importing batch 1 with 1 samples
Buffer resized from 22726bytes to 32529
Buffer resized from 32529bytes to 32693
Buffer resized from 32693bytes to 32738
Buffer resized from 32738bytes to 32741
Buffer resized from 32741bytes to 32756
Buffer resized from 32756bytes to 32768
Buffer resized from 32768bytes to 32769
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f7288295359, pid=68672, tid=0x00007f72dc187700
#
# JRE version: OpenJDK Runtime Environment (8.0_171-b10) (build 1.8.0_171-b10)
# Java VM: OpenJDK 64-Bit Server VM (25.171-b10 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libtiledbgenomicsdb8064358042335455262.so+0x155359]  BufferVariantCell::set_cell(void const*)+0x99
#
# Core dump written. Default location: /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/core or core.68672
#
# An error report file with more information is saved as:
# /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/hs_err_pid68672.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

```

And here is the `hs_err_pid68672.log` file: 
[hs_err_pid68672.log](https://github.com/broadinstitute/gatk/files/2219689/hs_err_pid68672.log)

### LINKED ISSUE #5300: Possible race condition in GenomicsDBImport multi-interval mode.
@vdauwera reports getting this error when running `GenomicsDBImport` with a large interval list as the `-L` input:

```
java.util.concurrent.CompletionException: org.broadinstitute.hellbender.exceptions.GATKException: Cannot call query with different interval, expected:1:29867-31003 queried with: 1:68590-70510
```

Looking at the code that produces this error, this seems like a "should never happen" type of error that would likely only be produced by a race condition of some kind.

Full stderr log follows:

```
Picked up _JAVA_OPTIONS: -Djava.io.tmpdir=/cromwell_root/tmp.876625c8
04:37:28.801 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/gatk/gatk-package-4.0.9.0-local.jar!/com/intel/gkl/native/libgkl_compression.so
04:37:29.319 INFO  GenomicsDBImport - ------------------------------------------------------------
04:37:29.319 INFO  GenomicsDBImport - The Genome Analysis Toolkit (GATK) v4.0.9.0
04:37:29.320 INFO  GenomicsDBImport - For support and documentation go to https://software.broadinstitute.org/gatk/
04:37:29.320 INFO  GenomicsDBImport - Executing as root@7a7880aef99b on Linux v4.9.0-0.bpo.6-amd64 amd64
04:37:29.321 INFO  GenomicsDBImport - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_171-8u171-b11-0ubuntu0.16.04.1-b11
04:37:29.321 INFO  GenomicsDBImport - Start Date/Time: October 8, 2018 4:37:28 AM UTC
04:37:29.321 INFO  GenomicsDBImport - ------------------------------------------------------------
04:37:29.322 INFO  GenomicsDBImport - ------------------------------------------------------------
04:37:29.322 INFO  GenomicsDBImport - HTSJDK Version: 2.16.1
04:37:29.323 INFO  GenomicsDBImport - Picard Version: 2.18.13
04:37:29.323 INFO  GenomicsDBImport - HTSJDK Defaults.COMPRESSION_LEVEL : 2
04:37:29.323 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false
04:37:29.324 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true
04:37:29.324 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false
04:37:29.324 INFO  GenomicsDBImport - Deflater: IntelDeflater
04:37:29.324 INFO  GenomicsDBImport - Inflater: IntelInflater
04:37:29.325 INFO  GenomicsDBImport - GCS max retries/reopens: 20
04:37:29.325 INFO  GenomicsDBImport - Requester pays: disabled
04:37:29.325 INFO  GenomicsDBImport - Initializing engine
04:37:29.993 INFO  FeatureManager - Using codec BEDCodec to read file file:///cromwell_root/gatk-test-data/intervals/Broad.human.exome.scattered/Broad.human.exome.b37_1_nozero.bed
04:37:30.330 INFO  IntervalArgumentCollection - Processing 18455956 bp from intervals
04:37:30.340 WARN  GenomicsDBImport - A large number of intervals were specified. Using more than 100 intervals in a single import is not recommended and can cause performance to suffer. It is recommended that intervals be aggregated together.
04:37:30.380 INFO  GenomicsDBImport - Done initializing engine
Created workspace /cromwell_root/genomicsdb
04:37:30.677 INFO  GenomicsDBImport - Vid Map JSON file will be written to genomicsdb/vidmap.json
04:37:30.677 INFO  GenomicsDBImport - Callset Map JSON file will be written to genomicsdb/callset.json
04:37:30.677 INFO  GenomicsDBImport - Complete VCF Header will be written to genomicsdb/vcfheader.vcf
04:37:30.678 INFO  GenomicsDBImport - Importing to array - genomicsdb/genomicsdb_array
04:37:30.680 INFO  ProgressMeter - Starting traversal
04:37:30.680 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Batches Processed   Batches/Minute
04:37:33.253 INFO  GenomicsDBImport - Starting batch input file preload
04:37:35.079 INFO  GenomicsDBImport - Finished batch preload
04:37:35.079 INFO  GenomicsDBImport - Importing batch 1 with 50 samples
04:37:37.079 INFO  GenomicsDBImport - Starting batch input file preload
04:37:38.712 INFO  GenomicsDBImport - Finished batch preload
04:37:38.712 INFO  GenomicsDBImport - Importing batch 1 with 50 samples
04:37:39.162 INFO  GenomicsDBImport - Shutting down engine
[October 8, 2018 4:37:39 AM UTC] org.broadinstitute.hellbender.tools.genomicsdb.GenomicsDBImport done. Elapsed time: 0.17 minutes.
Runtime.totalMemory()=4116185088
java.util.concurrent.CompletionException: org.broadinstitute.hellbender.exceptions.GATKException: Cannot call query with different interval, expected:1:29867-31003 queried with: 1:68590-70510
    at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
    at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
    at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1592)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)
Caused by: org.broadinstitute.hellbender.exceptions.GATKException: Cannot call query with different interval, expected:1:29867-31003 queried with: 1:68590-70510
    at org.broadinstitute.hellbender.tools.genomicsdb.GenomicsDBImport$InitializedQueryWrapper.query(GenomicsDBImport.java:769)
    at com.intel.genomicsdb.importer.GenomicsDBImporter.<init>(GenomicsDBImporter.java:165)
    at com.intel.genomicsdb.importer.GenomicsDBImporter.lambda$null$2(GenomicsDBImporter.java:604)
    at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
    ... 3 more
04:37:39.167 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.169 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.169 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.169 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.170 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.170 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.171 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.171 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.171 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.171 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.172 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.172 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.172 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.173 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.173 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.173 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.173 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.174 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.174 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.174 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.174 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.175 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.175 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.175 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.176 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.177 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.177 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.178 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.179 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.180 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.180 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.180 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.180 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.181 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.181 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.181 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.181 INFO  GenomicsDBImport - Starting batch input file preload
04:37:39.187 INFO  GenomicsDBImport - Starting batch input file preload
Using GATK jar /gatk/gatk-package-4.0.9.0-local.jar
Running:
    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xmx4g -Xms4g -jar /gatk/gatk-package-4.0.9.0-local.jar GenomicsDBImport --genomicsdb-workspace-path genomicsdb --batch-size 50 -L /cromwell_root/gatk-test-data/intervals/Broad.human.exome.scattered/Broad.human.exome.b37_1_nozero.bed --sample-name-map inputs.list --reader-threads 5 -ip 500
```

### PR DESCRIPTION
**Pull request by kgururaj**
The newest release of GenomicsDB treats spanning deletions (spanning
from earlier positions) as deletions in the min PL value computation.
This behavior now matches the behavior of CombineGVCFs.

A more detailed description of the issue is provided in
https://github.com/broadinstitute/gatk/pull/4963

* Deleted a couple of files which are no longer necessary.
* Fixed the index of newMQcalc.combined.g.vcf
* Fixes #5045 (error out with a helpful error message)
* Fixes #5300

### CODE REVIEW

**Comment by kgururaj** [comment]
@nalinigans  - this should fix the tests with respect to the latest jar

**Comment by kgururaj** [comment]
This release also errors out with a descriptive error message if the length of a field in the data lines does not match the length descriptor in the header - see https://github.com/broadinstitute/gatk/issues/5045.
Error out behavior as per [Laura's comment here](https://github.com/broadinstitute/gatk/issues/5113#issuecomment-413667356)

File: src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java [hunk 1]
```diff
@@ -810,10 +810,11 @@ private InitializedQueryWrapper(final FeatureReader<VariantContext> reader, fina
         }
 
         @Override
-        public CloseableTribbleIterator<VariantContext> query(final String chr, final int start, final int end) {
+        public CloseableTribbleIterator<VariantContext> query(final String chr, final int start, final int end) throws IOException {
             final SimpleInterval queryInterval = new SimpleInterval(chr, start, end);
             if( !interval.equals(queryInterval)){
-                throw new GATKException("Cannot call query with different interval, expected:" + this.interval + " queried with: " + queryInterval);
+                this.interval = new SimpleInterval(chr, start, end);
+                this.query = reader.query(interval.getContig(), interval.getStart(), interval.getEnd());
```
Comment (kgururaj) [inline src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java]: @droazen @lbergelson would it be possible for one of you take a look at this fix for #5300? I think my fix preserves the original intent of InitializedQueryWrapper which appears to be to speed up the opening of VCF readers. I didn't write this part of the code so I may not fully understand the objective.

**Comment by droazen** [comment]
@kgururaj Could you rebase to resolve conflicts on this branch?

**Comment by droazen** [comment]
@kgururaj Pinging you on this one -- once the review comments above are addressed and the branch is rebased, I think this is good to go.

**Comment by droazen** [comment]
Since @kgururaj is away, I'll address the remaining comments on this PR myself, since they are so minor.

**Comment by droazen** [comment]
@nalinigans @kgururaj I addressed the review comments and rebased the branch, however after rebasing some of the GenomicsDB tests that assert an exact match against `CombineGVCFs` output are failing, despite my regenerating the expected output using the latest `CombineGVCFs`.

See the commit https://github.com/broadinstitute/gatk/pull/5397/commits/500b0263fd51bfb125297d5a85a9e58ce9278682, where I've added comments to the failing tests `testGenomicsDBImportFileInputs_newMQ()` and `testGenomicsDBImportFileInputsAgainstCombineGVCFWithNonDiploidData()`

Could one of you take a look at these failures and give your opinion? Is it possible that GenomicsDB no longer matches `CombineGVCFs` in these cases?

I found a note in the recent commit https://github.com/broadinstitute/gatk/commit/fb701919bc618bb5be94c1c85b69a73924842443 (from the PR https://github.com/broadinstitute/gatk/pull/5471) in which the author found it necessary to make separate copies of the `HaplotypeCaller` expected GVCFs in order to get the GenomicsDB tests to pass. This is likely related to the test failures we're seeing now.

**Comment by droazen** [comment]
Could also possibly be related to https://github.com/broadinstitute/gatk/issues/5160?

**Comment by droazen** [comment]
Hmm, that's strange, now only `testGenomicsDBImportFileInputs_newMQ()` is failing. `testGenomicsDBImportFileInputsAgainstCombineGVCFWithNonDiploidData()`, which was failing for me this morning, is now passing despite no additional changes.

**Comment by nalinigans** [comment]
Got the tests to pass by generating new expected result as mentioned by @kgururaj comment for expected.testGenomicsDBImportWithNonDiploidData.vcf and by using the GenomicsDBImports folder for expected results as some of the vcf files were updated as part of PR 5170.

* testGenomicsDBImportFileInputsAgainstCombineGVCFWithNonDiploidData
* testGenomicsDBImportFileInputs_newMQ

@droazen, please feel free to merge. Thanks.

File: src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java [hunk 2]
```diff
@@ -491,11 +491,18 @@ public void onTraversalStart() {
...
+            if( intervals.size() == 1) {
+                final ThreadFactory threadFactory = new ThreadFactoryBuilder()
                     .setNameFormat("readerInitializer-thread-%d")
                     .setDaemon(true)
                     .build();
-            this.inputPreloadExecutorService = Executors.newFixedThreadPool(vcfInitializerThreads, threadFactory);
+                this.inputPreloadExecutorService = Executors.newFixedThreadPool(vcfInitializerThreads, threadFactory);
+            }
+            else {
+                logger.info("GenomicsDBImport cannot use multiple VCF reader threads for initialization when the "
```
[the remark this answers is not in the document]
Comment (nalinigans) [reply src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java]: Changed to logger.warn by @droazen.

File: src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java [hunk 3]
```diff
@@ -491,11 +491,18 @@ public void onTraversalStart() {
 
     private void initializeInputPreloadExecutorService() {
         if( vcfInitializerThreads > 1) {
```
[the remark this answers is not in the document]
Comment (nalinigans) [reply src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java]: Changed by @droazen

**Pull request merged**